src/librustdoc/clean/auto_trait.rs

```diff
@@ -99,29 +131,126 @@ impl<'a, 'tcx, 'rcx> AutoTraitFinder<'a, 'tcx, 'rcx> {
+                                    path: get_path_for_type(infcx.tcx,
+                                                            trait_def_id,
+                                                            hir::def::Def::Trait),
+                                    ref_id: ast::DUMMY_NODE_ID,
+                                };
+                                let provided_trait_methods =
+                                    infcx.tcx.provided_trait_methods(trait_def_id)
+                                             .into_iter()
+                                             .map(|meth| meth.ident.to_string())
+                                             .collect();
+
+                                let ty = self.get_real_ty(def_id, def_ctor, &real_name, generics);
+
+                                traits.push(Item {
+                                    source: Span::empty(),
+                                    name: None,
+                                    attrs: Default::default(),
+                                    visibility: None,
+                                    def_id: self.next_def_id(impl_def_id.krate),
+                                    stability: None,
+                                    deprecation: None,
+                                    inner: ImplItem(Impl {
+                                        unsafety: hir::Unsafety::Normal,
+                                        generics: (generics,
+                                                   &tcx.predicates_of(impl_def_id)).clean(self.cx),
+                                        provided_trait_methods,
+                                        trait_: Some(trait_.clean(self.cx)),
+                                        for_: ty.clean(self.cx),
+                                        items: infcx.tcx.associated_items(impl_def_id)
+                                                        .collect::<Vec<_>>()
+                                                        .clean(self.cx),
+                                        polarity: None,
+                                        synthetic: true,
+                                    }),
+                                });
+                                debug!("{:?} => {}", trait_ref, may_apply);
+                            }
+                        });
+                    });
+                }
+            }
+        }
+
         debug!(
             "get_auto_trait_impls(def_id={:?}, def_ctor=..., generics={:?}",
             def_id, generics
         );
-        let auto_traits: Vec<_> = self.cx
-            .send_trait
-            .and_then(|send_trait| {
-                self.get_auto_trait_impl_for(
-                    def_id,
-                    name.clone(),
-                    generics.clone(),
-                    def_ctor,
-                    send_trait,
-                )
-            })
-            .into_iter()
+        let auto_traits: Vec<_> =
+            self.cx.send_trait
```

Comment:
that's some funky formatting here. revert this block?

Reply:
😎

Reply:
Damn I thought I did! >< Might have been lost in my multiple tests...